cli/Sources/TuistHasher/TargetScriptsContentHasher.swift

```diff
@@ -32,9 +38,23 @@ public struct TargetScriptsContentHasher: TargetScriptsContentHashing {
             var pathsToHash: [AbsolutePath] = []
             script.path.map { pathsToHash.append($0) }
 
-            var dynamicPaths = (script.inputPaths + script.inputFileListPaths)
+            var dynamicPaths = script.inputPaths
                 .compactMap { try? AbsolutePath(validating: $0) }
                 .sorted()
+
+            let fileListPaths = script.inputFileListPaths
+                .compactMap { try? AbsolutePath(validating: $0) }
+                .sorted()
+
+            for fileListPath in fileListPaths {
+                await processInputFileListPath(
+                    fileListPath,
+                    sourceRootPath: sourceRootPath,
+                    pathsToHash: &pathsToHash,
+                    stringsToHash: &stringsToHash
```

Comment:
I'm not a fan of using `inout`, especially in cases like this when it provides very little value over returning a value. Why did you go for `inout`?

Additionally, the file list can contain a lot of files, so I'd consider parallelizing the process since we're doing the file-based operations